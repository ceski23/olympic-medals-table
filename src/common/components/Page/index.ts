export { Page } from "./Page";
export type { CountryMedalsData } from "./Page";
