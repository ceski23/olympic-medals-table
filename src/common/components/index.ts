export * from "./Select";
export * from "./Input";
export * from "./Page";
export * from "./App";
