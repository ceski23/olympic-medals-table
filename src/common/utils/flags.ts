export const flags = Object.fromEntries(
  Object.entries(
    import.meta.glob("../../../node_modules/country-flag-icons/3x2/*.svg", {
      as: "url",
      eager: true,
    })
  ).map(([id, url]) => [id.match(/([A-Z][A-Z])\.svg/)?.[1] ?? id, url])
);
