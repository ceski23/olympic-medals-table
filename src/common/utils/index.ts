export * from "./guards";
export * from "./styles";
export * from "./flags";
