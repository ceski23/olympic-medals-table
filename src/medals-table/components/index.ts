export * from "./CountryFlag";
export * from "./MedalsTable";
export * from "./SortingIndicator";
export * from "./Table";
export * from "./CountryForm";
