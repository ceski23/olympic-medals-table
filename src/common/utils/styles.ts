import { Theme } from "@emotion/react";
import { Breakpoint } from "common/styles";

export const mediaQueryUp =
  (breakpoint: Breakpoint) =>
  ({ theme }: { theme: Theme }) =>
    `@media (min-width: ${theme.breakpoints[breakpoint] + 0.01}px)`;

export const mediaQueryUpTo =
  (breakpoint: Breakpoint) =>
  ({ theme }: { theme: Theme }) =>
    `@media (max-width: ${theme.breakpoints[breakpoint]}px)`;
