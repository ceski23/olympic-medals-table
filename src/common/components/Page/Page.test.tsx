import { Page } from "common/components";
import { renderWithProviders } from "test/utils";

describe("Page", () => {
  it("should render correctly", () => {
    const { asFragment } = renderWithProviders(<Page />);
    expect(asFragment()).toMatchSnapshot();
  });
});
