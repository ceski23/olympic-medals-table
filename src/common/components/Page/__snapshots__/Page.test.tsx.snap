// Vitest Snapshot v1

exports[`Page > should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  gap: 50px;
}

@media (min-width: 768.01px) {
  .emotion-0 {
    padding: 50px;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 20px;
  width: 100%;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media (min-width: 1024.01px) {
  .emotion-2 {
    -webkit-box-flex-wrap: nowrap;
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
  }
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 100%;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

@media (min-width: 1024.01px) {
  .emotion-5 {
    min-width: 0;
  }
}

.emotion-7 {
  font-size: 0.9rem;
  margin-bottom: 5px;
  color: #707070;
}

.emotion-9 {
  border-radius: 5px;
  border: 1px solid #b6b6b6;
  padding: 10px 15px;
  font-size: 1rem;
  box-shadow: 3px 3px 7px #dddddd;
  min-width: 0;
}

.emotion-9 option:disabled {
  color: #707070;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width: 1024.01px) {
  .emotion-12 {
    min-width: 0;
  }
}

.emotion-16 {
  border-radius: 5px;
  border: 1px solid #b6b6b6;
  padding: 10px 15px;
  font-size: 1rem;
  box-shadow: 3px 3px 7px #dddddd;
}

.emotion-32 {
  border-radius: 5px;
  border: none;
  padding: 10px 15px;
  background-color: #0084ff;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
  min-width: 100px;
  color: #ffffff;
  margin-bottom: auto;
  font-weight: 600;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin-top: 24px;
  text-transform: uppercase;
}

.emotion-32:hover {
  background-color: #49a7ff;
}

@media (min-width: 1024.01px) {
  .emotion-32 {
    -webkit-flex-basis: unset;
    -ms-flex-preferred-size: unset;
    flex-basis: unset;
  }
}

.emotion-34 {
  overflow-x: auto;
  border-radius: 5px;
  width: 100%;
  box-shadow: 0 10px 20px #dddddd;
  border: 1px solid #b6b6b6;
}

.emotion-36 {
  width: 100%;
  border-collapse: collapse;
  min-width: 500px;
}

.emotion-38 {
  background-color: #ffffff;
}

.emotion-40 {
  min-width: -webkit-min-content;
  min-width: -moz-min-content;
  min-width: min-content;
}

.emotion-42 {
  padding: 10px 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  gap: 10px;
}

.emotion-44 {
  width: 20px;
  height: 20px;
}

.emotion-74 {
  padding: 30px;
  background-color: #f6f6f6;
  color: #707070;
  text-align: center;
}

.emotion-76 {
  color: #707070;
}

.emotion-78 {
  background-color: #e9e9e9;
  border-radius: 3px;
  border: 1px solid #b6b6b6;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.2),0 2px 0 0 rgba(255, 255, 255, 0.7) inset;
  color: #000000;
  display: inline-block;
  font-size: 0.7rem;
  font-weight: 700;
  line-height: 1;
  padding: 2px 4px;
  white-space: nowrap;
  margin: 2px 4px;
}

<div
    class="emotion-0 emotion-1"
  >
    <form
      class="emotion-2 emotion-3"
      novalidate=""
    >
      <div
        class="emotion-4 emotion-5 emotion-6"
      >
        <label
          class="emotion-7 emotion-8"
          for="countrySelect"
        >
          Country
        </label>
        <select
          aria-invalid="false"
          class="emotion-9 emotion-10"
          id="countrySelect"
          name="country"
        >
          <option
            disabled=""
            selected=""
            value=""
          >
            Select country...
          </option>
          <option
            value="AND"
          >
            Andorra
          </option>
          <option
            value="ARE"
          >
            United Arab Emirates
          </option>
          <option
            value="AFG"
          >
            Afghanistan
          </option>
          <option
            value="ATG"
          >
            Antigua and Barbuda
          </option>
          <option
            value="AIA"
          >
            Anguilla
          </option>
          <option
            value="ALB"
          >
            Albania
          </option>
          <option
            value="ARM"
          >
            Armenia
          </option>
          <option
            value="AGO"
          >
            Angola
          </option>
          <option
            value="ATA"
          >
            Antarctica
          </option>
          <option
            value="ARG"
          >
            Argentina
          </option>
          <option
            value="ASM"
          >
            American Samoa
          </option>
          <option
            value="AUT"
          >
            Austria
          </option>
          <option
            value="AUS"
          >
            Australia
          </option>
          <option
            value="ABW"
          >
            Aruba
          </option>
          <option
            value="ALA"
          >
            Åland Islands
          </option>
          <option
            value="AZE"
          >
            Azerbaijan
          </option>
          <option
            value="BIH"
          >
            Bosnia and Herzegovina
          </option>
          <option
            value="BRB"
          >
            Barbados
          </option>
          <option
            value="BGD"
          >
            Bangladesh
          </option>
          <option
            value="BEL"
          >
            Belgium
          </option>
          <option
            value="BFA"
          >
            Burkina Faso
          </option>
          <option
            value="BGR"
          >
            Bulgaria
          </option>
          <option
            value="BHR"
          >
            Bahrain
          </option>
          <option
            value="BDI"
          >
            Burundi
          </option>
          <option
            value="BEN"
          >
            Benin
          </option>
          <option
            value="BLM"
          >
            Saint Barthélemy
          </option>
          <option
            value="BMU"
          >
            Bermuda
          </option>
          <option
            value="BRN"
          >
            Brunei Darussalam
          </option>
          <option
            value="BOL"
          >
            Bolivia
          </option>
          <option
            value="BES"
          >
            Bonaire, Sint Eustatius and Saba
          </option>
          <option
            value="BRA"
          >
            Brazil
          </option>
          <option
            value="BHS"
          >
            Bahamas
          </option>
          <option
            value="BTN"
          >
            Bhutan
          </option>
          <option
            value="BVT"
          >
            Bouvet Island
          </option>
          <option
            value="BWA"
          >
            Botswana
          </option>
          <option
            value="BLR"
          >
            Belarus
          </option>
          <option
            value="BLZ"
          >
            Belize
          </option>
          <option
            value="CAN"
          >
            Canada
          </option>
          <option
            value="CCK"
          >
            Cocos Islands
          </option>
          <option
            value="COD"
          >
            Congo
          </option>
          <option
            value="CAF"
          >
            Central African Republic
          </option>
          <option
            value="COG"
          >
            Congo
          </option>
          <option
            value="CHE"
          >
            Switzerland
          </option>
          <option
            value="CIV"
          >
            Côte d'Ivoire
          </option>
          <option
            value="COK"
          >
            Cook Islands
          </option>
          <option
            value="CHL"
          >
            Chile
          </option>
          <option
            value="CMR"
          >
            Cameroon
          </option>
          <option
            value="CHN"
          >
            China
          </option>
          <option
            value="COL"
          >
            Colombia
          </option>
          <option
            value="CRI"
          >
            Costa Rica
          </option>
          <option
            value="CUB"
          >
            Cuba
          </option>
          <option
            value="CPV"
          >
            Cabo Verde
          </option>
          <option
            value="CUW"
          >
            Curaçao
          </option>
          <option
            value="CXR"
          >
            Christmas Island
          </option>
          <option
            value="CYP"
          >
            Cyprus
          </option>
          <option
            value="CZE"
          >
            Czech Republic
          </option>
          <option
            value="DEU"
          >
            Germany
          </option>
          <option
            value="DJI"
          >
            Djibouti
          </option>
          <option
            value="DNK"
          >
            Denmark
          </option>
          <option
            value="DMA"
          >
            Dominica
          </option>
          <option
            value="DOM"
          >
            Dominican Republic
          </option>
          <option
            value="DZA"
          >
            Algeria
          </option>
          <option
            value="ECU"
          >
            Ecuador
          </option>
          <option
            value="EST"
          >
            Estonia
          </option>
          <option
            value="EGY"
          >
            Egypt
          </option>
          <option
            value="ESH"
          >
            Western Sahara
          </option>
          <option
            value="ERI"
          >
            Eritrea
          </option>
          <option
            value="ESP"
          >
            Spain
          </option>
          <option
            value="ETH"
          >
            Ethiopia
          </option>
          <option
            value="FIN"
          >
            Finland
          </option>
          <option
            value="FJI"
          >
            Fiji
          </option>
          <option
            value="FLK"
          >
            Falkland Islands
          </option>
          <option
            value="FSM"
          >
            Federated States of Micronesia
          </option>
          <option
            value="FRO"
          >
            Faroe Islands
          </option>
          <option
            value="FRA"
          >
            France
          </option>
          <option
            value="GAB"
          >
            Gabon
          </option>
          <option
            value="GBR"
          >
            United Kingdom of Great Britain and Northern Ireland
          </option>
          <option
            value="GRD"
          >
            Grenada
          </option>
          <option
            value="GEO"
          >
            Georgia
          </option>
          <option
            value="GUF"
          >
            French Guiana
          </option>
          <option
            value="GGY"
          >
            Guernsey
          </option>
          <option
            value="GHA"
          >
            Ghana
          </option>
          <option
            value="GIB"
          >
            Gibraltar
          </option>
          <option
            value="GRL"
          >
            Greenland
          </option>
          <option
            value="GMB"
          >
            Gambia
          </option>
          <option
            value="GIN"
          >
            Guinea
          </option>
          <option
            value="GLP"
          >
            Guadeloupe
          </option>
          <option
            value="GNQ"
          >
            Equatorial Guinea
          </option>
          <option
            value="GRC"
          >
            Greece
          </option>
          <option
            value="SGS"
          >
            South Georgia and the South Sandwich Islands
          </option>
          <option
            value="GTM"
          >
            Guatemala
          </option>
          <option
            value="GUM"
          >
            Guam
          </option>
          <option
            value="GNB"
          >
            Guinea-Bissau
          </option>
          <option
            value="GUY"
          >
            Guyana
          </option>
          <option
            value="HKG"
          >
            Hong Kong
          </option>
          <option
            value="HMD"
          >
            Heard Island and McDonald Islands
          </option>
          <option
            value="HND"
          >
            Honduras
          </option>
          <option
            value="HRV"
          >
            Croatia
          </option>
          <option
            value="HTI"
          >
            Haiti
          </option>
          <option
            value="HUN"
          >
            Hungary
          </option>
          <option
            value="IDN"
          >
            Indonesia
          </option>
          <option
            value="IRL"
          >
            Ireland
          </option>
          <option
            value="ISR"
          >
            Israel
          </option>
          <option
            value="IMN"
          >
            Isle of Man
          </option>
          <option
            value="IND"
          >
            India
          </option>
          <option
            value="IOT"
          >
            British Indian Ocean Territory
          </option>
          <option
            value="IRQ"
          >
            Iraq
          </option>
          <option
            value="IRN"
          >
            Islamic Republic of Iran
          </option>
          <option
            value="ISL"
          >
            Iceland
          </option>
          <option
            value="ITA"
          >
            Italy
          </option>
          <option
            value="JEY"
          >
            Jersey
          </option>
          <option
            value="JAM"
          >
            Jamaica
          </option>
          <option
            value="JOR"
          >
            Jordan
          </option>
          <option
            value="JPN"
          >
            Japan
          </option>
          <option
            value="KEN"
          >
            Kenya
          </option>
          <option
            value="KGZ"
          >
            Kyrgyzstan
          </option>
          <option
            value="KHM"
          >
            Cambodia
          </option>
          <option
            value="KIR"
          >
            Kiribati
          </option>
          <option
            value="COM"
          >
            Comoros
          </option>
          <option
            value="KNA"
          >
            Saint Kitts and Nevis
          </option>
          <option
            value="PRK"
          >
            Democratic People's Republic of Korea
          </option>
          <option
            value="KOR"
          >
            Republic of Korea
          </option>
          <option
            value="KWT"
          >
            Kuwait
          </option>
          <option
            value="CYM"
          >
            Cayman Islands
          </option>
          <option
            value="KAZ"
          >
            Kazakhstan
          </option>
          <option
            value="LAO"
          >
            Lao People's Democratic Republic
          </option>
          <option
            value="LBN"
          >
            Lebanon
          </option>
          <option
            value="LCA"
          >
            Saint Lucia
          </option>
          <option
            value="LIE"
          >
            Liechtenstein
          </option>
          <option
            value="LKA"
          >
            Sri Lanka
          </option>
          <option
            value="LBR"
          >
            Liberia
          </option>
          <option
            value="LSO"
          >
            Lesotho
          </option>
          <option
            value="LTU"
          >
            Lithuania
          </option>
          <option
            value="LUX"
          >
            Luxembourg
          </option>
          <option
            value="LVA"
          >
            Latvia
          </option>
          <option
            value="LBY"
          >
            Libya
          </option>
          <option
            value="MAR"
          >
            Morocco
          </option>
          <option
            value="MCO"
          >
            Monaco
          </option>
          <option
            value="MDA"
          >
            Republic of Moldova
          </option>
          <option
            value="MNE"
          >
            Montenegro
          </option>
          <option
            value="MAF"
          >
            Saint Martin
          </option>
          <option
            value="MDG"
          >
            Madagascar
          </option>
          <option
            value="MHL"
          >
            Marshall Islands
          </option>
          <option
            value="MKD"
          >
            Macedonia
          </option>
          <option
            value="MLI"
          >
            Mali
          </option>
          <option
            value="MMR"
          >
            Myanmar
          </option>
          <option
            value="MNG"
          >
            Mongolia
          </option>
          <option
            value="MAC"
          >
            Macao
          </option>
          <option
            value="MNP"
          >
            Northern Mariana Islands
          </option>
          <option
            value="MTQ"
          >
            Martinique
          </option>
          <option
            value="MRT"
          >
            Mauritania
          </option>
          <option
            value="MSR"
          >
            Montserrat
          </option>
          <option
            value="MLT"
          >
            Malta
          </option>
          <option
            value="MUS"
          >
            Mauritius
          </option>
          <option
            value="MDV"
          >
            Maldives
          </option>
          <option
            value="MWI"
          >
            Malawi
          </option>
          <option
            value="MEX"
          >
            Mexico
          </option>
          <option
            value="MYS"
          >
            Malaysia
          </option>
          <option
            value="MOZ"
          >
            Mozambique
          </option>
          <option
            value="NAM"
          >
            Namibia
          </option>
          <option
            value="NCL"
          >
            New Caledonia
          </option>
          <option
            value="NER"
          >
            Niger
          </option>
          <option
            value="NFK"
          >
            Norfolk Island
          </option>
          <option
            value="NGA"
          >
            Nigeria
          </option>
          <option
            value="NIC"
          >
            Nicaragua
          </option>
          <option
            value="NLD"
          >
            Netherlands
          </option>
          <option
            value="NOR"
          >
            Norway
          </option>
          <option
            value="NPL"
          >
            Nepal
          </option>
          <option
            value="NRU"
          >
            Nauru
          </option>
          <option
            value="NIU"
          >
            Niue
          </option>
          <option
            value="NZL"
          >
            New Zealand
          </option>
          <option
            value="OMN"
          >
            Oman
          </option>
          <option
            value="PAN"
          >
            Panama
          </option>
          <option
            value="PER"
          >
            Peru
          </option>
          <option
            value="PYF"
          >
            French Polynesia
          </option>
          <option
            value="PNG"
          >
            Papua New Guinea
          </option>
          <option
            value="PHL"
          >
            Philippines
          </option>
          <option
            value="PAK"
          >
            Pakistan
          </option>
          <option
            value="POL"
          >
            Poland
          </option>
          <option
            value="SPM"
          >
            Saint Pierre and Miquelon
          </option>
          <option
            value="PCN"
          >
            Pitcairn
          </option>
          <option
            value="PRI"
          >
            Puerto Rico
          </option>
          <option
            value="PSE"
          >
            State of Palestine
          </option>
          <option
            value="PRT"
          >
            Portugal
          </option>
          <option
            value="PLW"
          >
            Palau
          </option>
          <option
            value="PRY"
          >
            Paraguay
          </option>
          <option
            value="QAT"
          >
            Qatar
          </option>
          <option
            value="REU"
          >
            Réunion
          </option>
          <option
            value="ROU"
          >
            Romania
          </option>
          <option
            value="SRB"
          >
            Serbia
          </option>
          <option
            value="RUS"
          >
            Russian Federation
          </option>
          <option
            value="RWA"
          >
            Rwanda
          </option>
          <option
            value="SAU"
          >
            Saudi Arabia
          </option>
          <option
            value="SLB"
          >
            Solomon Islands
          </option>
          <option
            value="SYC"
          >
            Seychelles
          </option>
          <option
            value="SDN"
          >
            Sudan
          </option>
          <option
            value="SWE"
          >
            Sweden
          </option>
          <option
            value="SGP"
          >
            Singapore
          </option>
          <option
            value="SHN"
          >
            Saint Helena, Ascension and Tristan da Cunha
          </option>
          <option
            value="SVN"
          >
            Slovenia
          </option>
          <option
            value="SJM"
          >
            Svalbard and Jan Mayen
          </option>
          <option
            value="SVK"
          >
            Slovakia
          </option>
          <option
            value="SLE"
          >
            Sierra Leone
          </option>
          <option
            value="SMR"
          >
            San Marino
          </option>
          <option
            value="SEN"
          >
            Senegal
          </option>
          <option
            value="SOM"
          >
            Somalia
          </option>
          <option
            value="SUR"
          >
            Suriname
          </option>
          <option
            value="SSD"
          >
            South Sudan
          </option>
          <option
            value="STP"
          >
            Sao Tome and Principe
          </option>
          <option
            value="SLV"
          >
            El Salvador
          </option>
          <option
            value="SXM"
          >
            Sint Maarten
          </option>
          <option
            value="SYR"
          >
            Syrian Arab Republic
          </option>
          <option
            value="SWZ"
          >
            Swaziland
          </option>
          <option
            value="TCA"
          >
            Turks and Caicos Islands
          </option>
          <option
            value="TCD"
          >
            Chad
          </option>
          <option
            value="ATF"
          >
            French Southern Territories
          </option>
          <option
            value="TGO"
          >
            Togo
          </option>
          <option
            value="THA"
          >
            Thailand
          </option>
          <option
            value="TJK"
          >
            Tajikistan
          </option>
          <option
            value="TKL"
          >
            Tokelau
          </option>
          <option
            value="TLS"
          >
            Timor-Leste
          </option>
          <option
            value="TKM"
          >
            Turkmenistan
          </option>
          <option
            value="TUN"
          >
            Tunisia
          </option>
          <option
            value="TON"
          >
            Tonga
          </option>
          <option
            value="TUR"
          >
            Turkey
          </option>
          <option
            value="TTO"
          >
            Trinidad and Tobago
          </option>
          <option
            value="TUV"
          >
            Tuvalu
          </option>
          <option
            value="TWN"
          >
            Taiwan, Province of China
          </option>
          <option
            value="TZA"
          >
            United Republic of Tanzania
          </option>
          <option
            value="UKR"
          >
            Ukraine
          </option>
          <option
            value="UGA"
          >
            Uganda
          </option>
          <option
            value="UMI"
          >
            United States Minor Outlying Islands
          </option>
          <option
            value="USA"
          >
            United States of America
          </option>
          <option
            value="URY"
          >
            Uruguay
          </option>
          <option
            value="UZB"
          >
            Uzbekistan
          </option>
          <option
            value="VAT"
          >
            Holy See
          </option>
          <option
            value="VCT"
          >
            Saint Vincent and the Grenadines
          </option>
          <option
            value="VEN"
          >
            Venezuela (Bolivarian Republic of)
          </option>
          <option
            value="VGB"
          >
            Virgin Islands
          </option>
          <option
            value="VIR"
          >
            Virgin Islands of the United States
          </option>
          <option
            value="VNM"
          >
            Viet Nam
          </option>
          <option
            value="VUT"
          >
            Vanuatu
          </option>
          <option
            value="WLF"
          >
            Wallis and Futuna
          </option>
          <option
            value="WSM"
          >
            Samoa
          </option>
          <option
            value="YEM"
          >
            Yemen
          </option>
          <option
            value="MYT"
          >
            Mayotte
          </option>
          <option
            value="ZAF"
          >
            South Africa
          </option>
          <option
            value="ZMB"
          >
            Zambia
          </option>
          <option
            value="ZWE"
          >
            Zimbabwe
          </option>
        </select>
      </div>
      <div
        class="emotion-11 emotion-12 emotion-13"
      >
        <label
          class="emotion-7 emotion-15"
          for="goldMedalsInput"
        >
          Gold medals
        </label>
        <input
          aria-invalid="false"
          class="emotion-16 emotion-17"
          id="goldMedalsInput"
          inputmode="numeric"
          name="medals.gold"
          type="number"
        />
      </div>
      <div
        class="emotion-11 emotion-12 emotion-13"
      >
        <label
          class="emotion-7 emotion-15"
          for="silverMedalsInput"
        >
          Silver medals
        </label>
        <input
          aria-invalid="false"
          class="emotion-16 emotion-17"
          id="silverMedalsInput"
          inputmode="numeric"
          name="medals.silver"
          type="number"
        />
      </div>
      <div
        class="emotion-11 emotion-12 emotion-13"
      >
        <label
          class="emotion-7 emotion-15"
          for="bronzeMedalsInput"
        >
          Bronze medals
        </label>
        <input
          aria-invalid="false"
          class="emotion-16 emotion-17"
          id="bronzeMedalsInput"
          inputmode="numeric"
          name="medals.bronze"
          type="number"
        />
      </div>
      <button
        class="emotion-32 emotion-33"
        type="submit"
      >
        add
      </button>
    </form>
    <div
      class="emotion-34 emotion-35"
    >
      <table
        class="emotion-36 emotion-37"
      >
        <thead
          class="emotion-38 emotion-39"
        >
          <tr>
            <th
              class="emotion-40 emotion-41"
            >
              <div
                class="emotion-42 emotion-43"
              >
                Country
                <svg
                  class="emotion-44 emotion-45"
                  viewBox="0 0 312.85 385.74"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18.26 174.61H294.6c16.23 0 24.355-19.617 12.879-31.094L169.309 5.336c-7.113-7.113-18.645-7.113-25.762 0L5.377 143.516C-6.092 154.993 2.033 174.61 18.26 174.61z"
                    fill="#dddddd"
                  />
                  <path
                    d="M5.38 242.23l138.17 138.18c7.113 7.11 18.648 7.11 25.762 0l138.17-138.18c11.477-11.473 3.352-31.094-12.879-31.094H18.253c-16.227 0-24.352 19.621-12.879 31.094z"
                    fill="#dddddd"
                  />
                </svg>
              </div>
            </th>
            <th
              class="emotion-40 emotion-41"
            >
              <div
                class="emotion-42 emotion-43"
              >
                <img
                  alt="Gold medals"
                  height="30px"
                  src="/src/assets/gold-medal.svg"
                  width="30px"
                />
                <svg
                  class="emotion-44 emotion-45"
                  viewBox="0 0 312.85 385.74"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18.26 174.61H294.6c16.23 0 24.355-19.617 12.879-31.094L169.309 5.336c-7.113-7.113-18.645-7.113-25.762 0L5.377 143.516C-6.092 154.993 2.033 174.61 18.26 174.61z"
                    fill="#dddddd"
                  />
                  <path
                    d="M5.38 242.23l138.17 138.18c7.113 7.11 18.648 7.11 25.762 0l138.17-138.18c11.477-11.473 3.352-31.094-12.879-31.094H18.253c-16.227 0-24.352 19.621-12.879 31.094z"
                    fill="#dddddd"
                  />
                </svg>
              </div>
            </th>
            <th
              class="emotion-40 emotion-41"
            >
              <div
                class="emotion-42 emotion-43"
              >
                <img
                  alt="Silver medals"
                  height="30px"
                  src="/src/assets/silver-medal.svg"
                  width="30px"
                />
                <svg
                  class="emotion-44 emotion-45"
                  viewBox="0 0 312.85 385.74"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18.26 174.61H294.6c16.23 0 24.355-19.617 12.879-31.094L169.309 5.336c-7.113-7.113-18.645-7.113-25.762 0L5.377 143.516C-6.092 154.993 2.033 174.61 18.26 174.61z"
                    fill="#dddddd"
                  />
                  <path
                    d="M5.38 242.23l138.17 138.18c7.113 7.11 18.648 7.11 25.762 0l138.17-138.18c11.477-11.473 3.352-31.094-12.879-31.094H18.253c-16.227 0-24.352 19.621-12.879 31.094z"
                    fill="#dddddd"
                  />
                </svg>
              </div>
            </th>
            <th
              class="emotion-40 emotion-41"
            >
              <div
                class="emotion-42 emotion-43"
              >
                <img
                  alt="Bronze medals"
                  height="30px"
                  src="/src/assets/bronze-medal.svg"
                  width="30px"
                />
                <svg
                  class="emotion-44 emotion-45"
                  viewBox="0 0 312.85 385.74"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18.26 174.61H294.6c16.23 0 24.355-19.617 12.879-31.094L169.309 5.336c-7.113-7.113-18.645-7.113-25.762 0L5.377 143.516C-6.092 154.993 2.033 174.61 18.26 174.61z"
                    fill="#dddddd"
                  />
                  <path
                    d="M5.38 242.23l138.17 138.18c7.113 7.11 18.648 7.11 25.762 0l138.17-138.18c11.477-11.473 3.352-31.094-12.879-31.094H18.253c-16.227 0-24.352 19.621-12.879 31.094z"
                    fill="#dddddd"
                  />
                </svg>
              </div>
            </th>
            <th
              class="emotion-40 emotion-41"
            >
              <div
                class="emotion-42 emotion-43"
              >
                Totals
                <svg
                  class="emotion-44 emotion-45"
                  viewBox="0 0 312.85 385.74"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18.26 174.61H294.6c16.23 0 24.355-19.617 12.879-31.094L169.309 5.336c-7.113-7.113-18.645-7.113-25.762 0L5.377 143.516C-6.092 154.993 2.033 174.61 18.26 174.61z"
                    fill="#dddddd"
                  />
                  <path
                    d="M5.38 242.23l138.17 138.18c7.113 7.11 18.648 7.11 25.762 0l138.17-138.18c11.477-11.473 3.352-31.094-12.879-31.094H18.253c-16.227 0-24.352 19.621-12.879 31.094z"
                    fill="#707070"
                  />
                </svg>
              </div>
            </th>
            <th
              class="emotion-40 emotion-41"
            >
              <div
                class="emotion-42 emotion-43"
              />
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="emptyPlaceholder"
          >
            <td
              class="emotion-74 emotion-75"
              colspan="6"
            >
              The table is empty
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="emotion-76 emotion-77"
    >
      Hint: Hold 
      <kbd
        class="emotion-78 emotion-79"
      >
        SHIFT
      </kbd>
       while sorting to use multisort!
    </div>
  </div>
</DocumentFragment>
`;
